Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/eventregistration/organizer/create/ApiRequestState.kt

```diff
@@ -0,0 +1,5 @@
+package de.rki.coronawarnapp.ui.eventregistration.organizer.create
+
+enum class ApiRequestState {
```

Comment:
What do you think about using a sealed class hierarchy here instead? Then we could add e.g. the exception to the FAILED and the uiState to SUCCESS and don't need additional LiveData properties to send this data to the Fragment.

Reply:
yeah, but the problem is that error is displayed only as a message so I'm using `SingleLiveEvent` not `MutableLiveData`. Also, the state of the network request is updated from another thread so in case I would want to implement it as part of `uiState` I would need to keep it as a separate variable out of `uiState` anyway to prevent accidental rewriting, so I'm not sure if another approach would save us anything.

Reply:
Keeping it separate is also okay.
Currently state management here looks a bit chaotic, could we make it nicer somehow?
For threadsafe updates from multiple sources you could look into our `HotDataFlow` class.

Reply:
Code is updated. I have API response as `SingleLiveEvent` and UI state as `LiveData` object and I moved loading state as part of UI state. Let me know if it's more clear now.